src/gun_http2.erl

```diff
@@ -352,14 +357,14 @@ maybe_ack_or_notify(State=#http2_state{reply_to=ReplyTo, socket=Socket,
 		_ ->
 			ok
 	end,
-	State.
+	{state, State}.
 
 data_frame(State0, StreamID, IsFin, Data, CookieStore0, EvHandler, EvHandlerState0) ->
 	case get_stream_by_id(State0, StreamID) of
 		Stream=#stream{tunnel=undefined} ->
 			{State, EvHandlerState} = data_frame1(State0,
 				StreamID, IsFin, Data, EvHandler, EvHandlerState0, Stream),
-			{State, CookieStore0, EvHandlerState};
+			{{state, State}, CookieStore0, EvHandlerState};
```

Comment:
That can't be right, `data_frame1` can return an error in the code you add in this PR.

Reply:
`data_frame1` updated to return state tuple.